org-ql.el

```diff
@@ -825,6 +825,47 @@ A and B are Org headline elements."
             (a-priority t)
             (b-priority nil)))))
 
+;;;; Functions/Occur
+
+(defun org-ql-occur (query &optional keep-previous callback)
+  "Make a compact tree which shows all matches of the query.
+
+QUERY is an `org-ql' query sexp (quoted, since this is a
+function).
+
+The tree will show the lines where the query matches, and any other context
+defined in `org-show-context-detail', which see.
+
+When optional argument KEEP-PREVIOUS is non-nil, highlighting and
+exposing done by a previous call to `org-occur' or `org-ql-occur'
+will be kept, to allow stacking of calls to this command.
+
+Optional argument CALLBACK can be a function of no argument.  In this case,
+it is called with point at the end of the match, match data being set
+accordingly.  Current match is shown only if the return value is non-nil.
+The function must neither move point nor alter narrowing."
+  (interactive (list (read-minibuffer "Query: ")))
+  (unless keep-previous
+    (org-remove-occur-highlights nil nil t))
+  ;; (push (cons regexp callback) org-occur-parameters)
```

Comment:
I notice that you didn't include this code from `org-occur`:

```el
    (goto-char (point-min))
    (when (or (not keep-previous)	    ; do not want to keep
              (not org-occur-highlights)) ; no previous matches
      ;; hide everything
      (org-overview))
```
Did you remove it for a particular reason?  I think calling `org-overview` is probably a good idea by default.

Reply:
I seem to have deleted the code by mistake. It should be restored.